File: frontend/src/features/products/services/brandService.ts
import apiClient from '@/services/api/apiClient';
import { BrandResponse, BrandRequest } from '../types/product';
import { ApiResponse } from '@/types/apiResponse';

const BRAND_API = '/v1/brands';

export const brandService = {
  /**
   * Get all brands list
   */
  async getAllBrands(): Promise<BrandResponse[]> {
    const response = await apiClient.get<ApiResponse<BrandResponse[]>>(BRAND_API);
    return response.data.data || [];
  },

  /**
   * Get brand by ID or slug
   */
  async getBrand(idOrSlug: string | number): Promise<BrandResponse> {
    const response = await apiClient.get<ApiResponse<BrandResponse>>(`${BRAND_API}/${idOrSlug}`);
    return response.data.data;
  },

  /**
   * Create a new brand (ADMIN only)
   */
  async createBrand(data: BrandRequest): Promise<BrandResponse> {
    const response = await apiClient.post<ApiResponse<BrandResponse>>(BRAND_API, data);
    return response.data.data;
  },

  /**
   * Update an existing brand (ADMIN only)
   */
  async updateBrand(brandId: number, data: BrandRequest): Promise<BrandResponse> {
    const response = await apiClient.put<ApiResponse<BrandResponse>>(`${BRAND_API}/${brandId}`, data);
    return response.data.data;
  },

  /**
   * Delete an existing brand (ADMIN only)
   */
  async deleteBrand(brandId: number): Promise<void> {
    await apiClient.delete<ApiResponse<void>>(`${BRAND_API}/${brandId}`);
  }
};

File: backend/src/main/java/com/nexoracommerce/brand/service/BrandService.java
package com.nexoracommerce.brand.service;

import com.nexoracommerce.brand.dto.request.BrandRequest;
import com.nexoracommerce.brand.dto.response.BrandResponse;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import java.util.List;

public interface BrandService {
    /**
     * Lấy danh sách tất cả thương hiệu
     */
    List<BrandResponse> getAllBrands();

    /**
     * Lấy chi tiết thương hiệu theo ID hoặc Slug
     */
    BrandResponse getBrand(String idOrSlug);

    /**
     * Lấy chi tiết thương hiệu theo ID
     */
    BrandResponse getBrandById(Long id);

    /**
     * Lấy chi tiết thương hiệu theo Slug
     */
    BrandResponse getBrandBySlug(String slug);

    /**
     * Tạo thương hiệu mới (Admin)
     */
    BrandResponse createBrand(BrandRequest request);

    /**
     * Cập nhật thương hiệu (Admin)
     */
    BrandResponse updateBrand(Long id, BrandRequest request);

    /**
     * Xóa thương hiệu (Admin)
     */
    void deleteBrand(Long id);

    /**
     * Lấy danh sách thương hiệu có phân trang
     */
    Page<BrandResponse> getBrandsPageable(Pageable pageable);
}
